src/rpc/blockchain.cpp

```diff
@@ -2474,8 +2474,9 @@ static RPCHelpMan scanblocks()
         } while (start_index != stop_block);
 
         ret.pushKV("from_height", start_block_height);
-        ret.pushKV("to_height", stop_block->nHeight);
+        ret.pushKV("to_height", start_index->nHeight); // start_index is always the last scanned block here
```

Comment:
For reference,`end_range->nHeight` would be a bit easier to read, but is not guaranteed to have a value.

Reply:
yeah, the user could have aborted the process at the very beginning.

maybe could change it to:
```
end_range ? end_range->nHeight : start_index->nHeight;
```
but I'm not so sure that this makes code easier to read.